core/jsAndWasmShared/src/Date.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.datetime
+
+/**
+ * Exposes the [Date API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) to Kotlin.
+ */
+public external class Date() {
```

Comment:
We definitely do not want to have `Date` class in `kotlinx.datetime` package with this API.

Reply:
Do you suggest to remove `Date` dependent API from Wasm target?

Reply:
I'm not sure about its use cases for WasmJs target, but for JS this change would be breaking and also contradicting the interop scenarios it was developed for.

Reply:
I pushed the variant how to escape this issue. Please take a look.